hudi-common/src/main/java/org/apache/hudi/common/schema/HoodieSchemaUtils.java

```diff
@@ -307,6 +325,31 @@ public static HoodieSchemaField createNewSchemaField(String name, HoodieSchema s
     return HoodieSchemaField.fromAvroField(avroField);
   }
 
+  /**
+   * Creates a new schema field with the specified properties, including field order.
+   * This is equivalent to HoodieAvroUtils.createNewSchemaField() but returns HoodieSchemaField.
+   *
+   * @param name         field name
+   * @param schema       field schema
+   * @param doc          field documentation (can be null)
+   * @param defaultValue default value (can be null)
+   * @param order        field order for sorting
+   * @return new HoodieSchemaField instance
+   * @throws IllegalArgumentException if name, schema, or order is null/empty
+   * @since 1.2.0
+   */
+  public static HoodieSchemaField createNewSchemaField(String name, HoodieSchema schema,
+                                                       String doc, Object defaultValue, HoodieFieldOrder order) {
+    ValidationUtils.checkArgument(name != null && !name.isEmpty(), "Field name cannot be null or empty");
+    ValidationUtils.checkArgument(schema != null, "Field schema cannot be null");
+    ValidationUtils.checkArgument(order != null, "Field order cannot be null");
+
+    // Delegate to HoodieAvroUtils
+    Schema.Field avroField = HoodieAvroUtils.createNewSchemaField(
+        name, schema.toAvroSchema(), doc, defaultValue, order.toAvroOrder());
+    return HoodieSchemaField.fromAvroField(avroField);
+  }
```

Comment:
Can `createNewSchemaField(String name, HoodieSchema schema, String doc, Object defaultValue)` reuse the logic here?

Reply:
@yihua aligned with ethan here we can keep this for now

Reply:
Given that this is the delegation, it is OK for now.